### DISCUSSION TITLE
Built-in automatic cookie consent?

### DISCUSSION THREAD

**Discussion by Cwpute** [plateaukao/einkbro#202] [Ideas]
Hi, do you think it would be possible to add a built-in cookie consent ? There are browser add-ons which already do that, it seems they work a bit like ad-blockers with some custom stuff in some cases.
Here is a good one for reference: https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies

Many websites hide their pages behind consent windows which are a pain to navigate through, moreso on eink screens which do not load as fast a traditional ones. This would help a great deal seemlessly navigating the web.

**Comment by plateaukao** [comment]
@Cwpute 
It looks like it needs to provide solutions for every websites; so I may not do similar features to EinkBro.
https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies/blob/master/src/data/js/6_cookieHandler.js

**Comment by Cwpute** [comment]
but isn't the adblock working the same way ? i mean: doesn't it require a remote list of things to block and not block ? or is it all already contained in the code of einkbro ?
wouldn't it be possible for einkbro to access this (up to date) list you're linking to when it needs to ?

i don't know myself, these are genuine questions from someone who isn't knowledgeable in that domain :p i'm just asking away.

**Reply by plateaukao** [reply]
adblock is a bit different. It's much simpler. It only needs a list of website domains to be  blocked from.
As for cookie consent scenario, each website uses different flags to track whether user already consent the question or not. So, in order to do so, in addition to know current website domain, the corresponding flag should be remembered; and in order to automatically consent the question for users, browser needs to inject the value (true, agree) to system, so some javascript codes also need to be injected too.
As you can see below, in the url you provided, it tracked different values for different websites.

```
    case "letssingit.com":
    case "lyricsbox.com":
      return ["cookieconsent=1111"];

    case "kontaktbazar.at":
    case "hoernews.de":
      return ["cookieconsent_status=dismiss"];

    case "pee.place":
    case "nearest.place":
    case "postboxmap.com":
      return ["gdprconsent=1"];
```

Originally, EinkBro hard-coded all the adblock content in the app. But recently, I will change to download the content from website and update it periodically. It's kind of a standard about how to keep track of adblock sites. But for skipping cookie consent, I am not aware there's a standard for this information yet.

**Reply by Cwpute** [reply]
Ok, thanks a lot for that explanation :) i'll dig a lettle deeper on the subject then.

**Comment by Cwpute** [comment]
I opened up a thread [on the Discussion section for the IDCAC addon](https://github.com/OhMyGuus/I-Still-Dont-Care-About-Cookies/discussions/3831) to see what were their thoughts regarding some sort of unified, normalized solution for cookie consent… but didn't get any answer or traction unfortunately.
If this topic is interesting to you, don't hesitate to participate and give your insight or testimony on the matter !
